src/Features/LanguageServer/Protocol/Handler/RequestHandlerExecutionQueue.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    internal class RequestHandlerExecutionQueue
+    {
+        private class QueueItem
+        {
+            public Func<Task> Callback { get; set; }
+            public RequestProcessingMode Type { get; set; }
+        }
+
+        private readonly AsyncQueue<QueueItem> _queue;
+
+        public RequestHandlerExecutionQueue()
+        {
+            _queue = new AsyncQueue<QueueItem>();
+            _ = ProcessQueueAsync();
```

Comment:
seems to be fire-forget, with no way to stop it, or track what its' doing (For testing purposes).

Reply:
Controversial viewpoint time! In Razor we typically like to build an actual thread for this type of work because it allows us to track where work is being dispatched from and have a little more finegrained control. That being said it does have its own drawbacks. Just throwing it out there 😄